File: Java/src/main/java/parrot/NorwegianBlueParrot.java
package parrot;

public class NorwegianBlueParrot extends Parrot {
    private static final double MAXIMUM_SPEED = 24.0;
    private final double voltage;

    public NorwegianBlueParrot(double voltage) {
        super();
        this.voltage = voltage;
    }

    @Override
    public double getSpeed() {
        return getBaseSpeed(voltage);
    }

    private double getBaseSpeed(double voltage) {
        return Math.min(MAXIMUM_SPEED, voltage * BASE_SPEED);
    }
}

File: Kotlin/src/main/kotlin/parrot/NorwegianBlueParrot.kt
package parrot

import kotlin.math.min

class NorwegianBlueParrot(voltage: Double) : Parrot() {
    val speed: Double = getBaseSpeed(voltage)

    private fun getBaseSpeed(voltage: Double): Double = min(MAXIMUM_SPEED, voltage * BASE_SPEED)

    companion object {
        const val MAXIMUM_SPEED = 24.0
    }
}